packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -445,6 +446,90 @@ describe("Runtime", () => {
 						assert(submittedOps[1].metadata?.batchId === undefined, "Expected no batchId (1)");
 					}
 				});
+
+			for (const skipSafetyFlushDuringProcessStack of [true, undefined]) {
+				it("Inbound (non-runtime) op triggers flush due to refSeq changing", async () => {
+					const submittedBatches: {
+						messages: IBatchMessage[];
+						referenceSequenceNumber: number;
+					}[] = [];
+
+					const mockContext = getMockContext({
+						settings: {
+							"Fluid.ContainerRuntime.DisableFlushBeforeProcess":
+								skipSafetyFlushDuringProcessStack,
+						},
+					});
+					(
```

Comment:
Consider using sinon fakes here for better readability.

Reply:
I tried, the way I did it was throwing because submitBatchFn isn't set on the mock.  But I like the reminder, I want to be using Sinon wherever possible